### PR TITLE
parse JSON emitted to stdout from libhoney transmissions correctly

### PR DESCRIPTION
**Pull request by robbkidd**
In service to #19.

Integration tested with:
- [x] Python Beeline - `transmission_impl=FileTransmission(output=sys.stdout)`
- [x] Node Beeline - `transmission: "stdout"`

### CODE REVIEW

**Comment by robbkidd** [comment]
✅  Confirmed working with Python and Node Beelines.

File: logsapi/server.go [hunk 1]
```diff
@@ -64,14 +64,22 @@ func handler(libhoneyClient *libhoney.Client) http.HandlerFunc {
...
+				var jsonRecord map[string]interface{}
+				err := json.Unmarshal([]byte(v), &jsonRecord)
 				if err != nil {
+					// not JSON; we'll treat this log entry as a timestamped string
 					event.Timestamp = parseMessageTimestamp(event, msg)
 					event.AddField("record", msg.Record)
 				} else {
-					event.Timestamp = parseFunctionTimestamp(msg, record)
-					event.Add(record)
+					// we've got JSON
```
Comment (paulosman) [inline logsapi/server.go:74]: you've got JSON

File: logsapi/server.go [hunk 2]
```diff
@@ -64,14 +64,22 @@ func handler(libhoneyClient *libhoney.Client) http.HandlerFunc {
...
+					event.Timestamp = parseFunctionTimestamp(msg, jsonRecord)
+					if jsonRecord["data"] != nil {
```
Comment (paulosman) [inline logsapi/server.go:76]: is it worth checking to see if this is a map? Something like:

    switch v := jsonRecord["data"].(type) {
    case map[string]interface{}:
             // data key contains a map, so add fields from it
             event.Add(jsonRecord["data"])
    default:
             // data is not a map, so just add the record as if it were a flat JSON object
             event.Add(jsonRecord)
    }

File: logsapi/server.go (hunk 1 above)
Comment (robbkidd) [reply logsapi/server.go:74]: You've got mail!

File: logsapi/server.go (hunk 2 above)
Comment (robbkidd) [reply logsapi/server.go:76]: Yes! That would be more better.

**Review by paulosman — APPROVED** [review]
:+1: I did think it needed another branch :P

**Comment by robbkidd** [comment]
Request changes: "not indented enough"

**Pull request merged**